### ISSUE TITLE
Unwanted dynamic import in generated CJS/IIFE bundle

### ISSUE THREAD

**Issue by gurupras** [egoist/tsup#420]
First, I should mention that this is my first time playing around with bundles/bundlers and I'm probably doing something wrong on my end. Sorry if this is a non-issue.

I'm trying to generate a CJS/IIFE bundle for Web Workers and I'm having some trouble using tsup for this purpose. When I generate the bundle and attempt to load the web worker, I get an error stating:

> Uncaught ReferenceError: require is not defined

I have two files:
**filter.worker.js**
```js
import { a, b } from './filter.js'
...
```
**filter.js**
```js
import { v4 as uuidv4 } from 'uuid'

export function a (...) { ... }
export function b (...) { ... }
```

My `tsup.config.js` is as follows:
```ts
import { defineConfig } from 'tsup'

export default defineConfig({
  entryPoints: [
    'filter.worker.js'
  ],
  outDir: './dist/',
  clean: true,
  format: ['cjs'],
  splitting: false,
  sourcemap: false,
  bundle: true,
  legacyOutput: true,
  watch: false
})
```

When I try the same with `esbuild`, the generated bundle does not have any `require` statements and `uuid.v4` is statically included in the resulting bundle.

What should I do to get rid of the dynamic `require` statement in the generated bundle?

**Comment by egoist** [comment]
modules specified in `dependencies` are not bundled: https://tsup.egoist.sh/#excluding-packages 

you can either fix this by moving `uuid` to `devDependencies` or change the `format` to `cjs` (requires tsup v5.2.1).

**Comment by gurupras** [comment]
I think I _am_ using `cjs` above and it didn't work as expected. Maybe I have to do _both_? 

This is what the bundle generated by tsup with `format: ['cjs']` looks like for me:
**dist/filter.worker.js**
```js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __markAsModule = (target) => __defProp(target, "__esModule", { value: true });
var __reExport = (target, module2, desc) => {
  if (module2 && typeof module2 === "object" || typeof module2 === "function") {
    for (let key of __getOwnPropNames(module2))
      if (!__hasOwnProp.call(target, key) && key !== "default")
        __defProp(target, key, { get: () => module2[key], enumerable: !(desc = __getOwnPropDesc(module2, key)) || desc.enumerable });
  }
  return target;
};
var __toModule = (module2) => {
  return __reExport(__markAsModule(__defProp(module2 != null ? __create(__getProtoOf(module2)) : {}, "default", module2 && module2.__esModule && "default" in module2 ? { get: () => module2.default, enumerable: true } : { value: module2, enumerable: true })), module2);
};

// node_modules/tsup/assets/cjs_shims.js
var importMetaUrlShim = typeof document === "undefined" ? new (require("url")).URL("file:" + __filename).href : document.currentScript && document.currentScript.src || new URL("main.js", document.baseURI).href;

// filter.js
var import_uuid = __toModule(require("uuid"));
...
...
```
That dynamic import on the last line is what is causing me some issues I think.

**Comment by egoist** [comment]
> you can either fix this by moving uuid to devDependencies or changing the format to cjs (requires tsup v5.2.1).

sorry that's a typo, it should be `iife` format

**Comment by gurupras** [comment]
Ah! Alright, thanks. I'll give that a shot.
I'm pretty sure this wasn't working in `v5.2.0` but it looks like this was fixed in `v5.2.1` a few hours ago.
